bigtable/integration_test.go

```diff
@@ -2142,6 +2144,63 @@ func TestIntegration_AdminBackup(t *testing.T) {
 	}
 }
 
+// Blackhole directpath address to test fallback.
+func TestIntegration_DirectPathFallback(t *testing.T) {
+	ctx := context.Background()
+	testEnv, _, _, table, _, cleanup, err := setupIntegration(ctx, t)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup()
+
+	if err := populatePresidentsGraph(table); err != nil {
+		t.Fatal(err)
+	}
+
+	// Precondition: wait for DirectPath to connect.
+	countEnough := exerciseDirectPath(ctx, testEnv, table /*blackholeDP = */, false)
+	if !countEnough {
+		t.Fatalf("Failed to observe RPCs over DirectPath")
+	}
+
+	// Enable the blackhole, which will prevent communication with grpclb and thus DirectPath.
+	blackholeOrAllowDirectPath(testEnv, t /*blackholeDP = */, true)
+	countEnough = exerciseDirectPath(ctx, testEnv, table /*blackholeDP = */, true)
+	if !countEnough {
+		t.Fatalf("Failed to fallback to CFE after blackhole DirectPath")
+	}
+
+	// Make sure that the client will start reading from IPv6 again by sending new requests and
+	// checking the injected IPv6 counter has been updated.
+	blackholeOrAllowDirectPath(testEnv, t /*blackholeDP = */, false)
+	countEnough = exerciseDirectPath(ctx, testEnv, table /*blackholeDP = */, false)
+	if !countEnough {
+		t.Fatalf("Failed to fallback to CFE after blackhole DirectPath")
+	}
+}
+
+func exerciseDirectPath(ctx context.Context, testEnv IntegrationEnv, table *Table, isBlackhole bool) bool {
```

Comment:
It's unclear to me what the return value is. Perhaps a different function name?

Reply:
I changed the function name to `examineTraffic`. The function checks and returns if enough DirectPath or CFE traffic is seen within 2 minutes.